Resources/Private/JavaScript/InspectorEditors/Reference/index.js

```diff
@@ -0,0 +1,65 @@
+import React, {Component, PropTypes} from 'react';
+import SelectBox from '@neos-project/react-ui-components/lib/SelectBox/';
+import {connect} from 'react-redux';
+import {$transform, $get} from 'plow-js';
+
+const {ApiEndpoints} = window['@Neos:HostPluginAPI'];
+const {searchNodes} = ApiEndpoints;
+
+@connect($transform({
+    activeWorkspace: $get('cr.workspaces.active'),
+    siteNodePath: $get('cr.nodes.siteNode')
+}))
+export default class ReferenceEditor extends Component {
+
+    static propTypes = {
+        commit: PropTypes.func.isRequired,
+        value: PropTypes.any.isRequired,
+
+        options: PropTypes.shape({
+            nodeTypes: PropTypes.arrayOf(PropTypes.string)
+        }),
+
+        activeWorkspace: PropTypes.string.isRequired,
+        siteNodePath: PropTypes.string.isRequired
+    };
+
+    render() {
+        const {commit} = this.props;
+        const valueSoFar = this.props.value;
+
+        const options = ({value, callback}) => {
+            const searchQuery = {
+                // TODO: dimensons!
+                workspaceName: this.props.activeWorkspace,
```

Comment:
Does the editor need to be aware of all this low-level stuff like dimensions and workspaces? Maybe move it to `searchNodes` API?